api/app/auto_spatial_advisory/process_hfi.py

```diff
@@ -98,27 +98,38 @@ async def process_hfi(run_type: RunType, run_date: date, run_datetime: datetime,
-        # within the past 7 days, don't apply an old snow mask, work with the classified hfi data as is
-        if last_processed_snow is None or last_processed_snow[0].for_date + datetime.timedelta(days=7) < datetime.now():
-            logger.info("No recently processed snow data found. Proceeding with non-masked hfi data.")
-            working_hfi_path = temp_filename
-        else:
-            working_hfi_path = await apply_snow_mask(temp_filename, last_processed_snow[0], temp_dir)
-        # Create a snow coverage mask from previously downloaded snow data.
-        with polygonize_in_memory(working_hfi_path, "hfi", "hfi") as layer:
-            # We need a geojson file to pass to tippecanoe
-            temp_geojson = write_geojson(layer, temp_dir)
-
-            pmtiles_filename = f'hfi{for_date.strftime("%Y%m%d")}.pmtiles'
-            temp_pmtiles_filepath = os.path.join(temp_dir, pmtiles_filename)
-            logger.info(f"Writing pmtiles -- {pmtiles_filename}")
-            tippecanoe_wrapper(temp_geojson, temp_pmtiles_filepath, min_zoom=HFI_PMTILES_MIN_ZOOM, max_zoom=HFI_PMTILES_MAX_ZOOM)
-
-            async with get_client() as (client, bucket):
+    async with get_client() as (client, bucket):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_filename = os.path.join(temp_dir, "classified.tif")
+            classify_hfi(hfi_key, temp_filename)
+            # If something has gone wrong with the collection of snow coverage data and it has not been collected
+            # within the past 7 days, don't apply an old snow mask, work with the classified hfi data as is
+            if last_processed_snow is None or last_processed_snow[0].for_date + datetime.timedelta(days=7) < datetime.now():
+                logger.info("No recently processed snow data found. Proceeding with non-masked hfi data.")
+                working_hfi_path = temp_filename
+            else:
+                # Create a snow coverage mask from previously downloaded snow data.
+                working_hfi_path = await apply_snow_mask(temp_filename, last_processed_snow[0], temp_dir)
+
+            raster_filename = get_raster_tif_filename(for_date)
+            raster_key = get_raster_filepath(run_date, run_type, raster_filename)
+            logger.info(f"Uploading file {raster_filename} to {raster_key}")
+            await client.put_object(
+                Bucket=bucket,
+                Key=raster_key,
+                ACL=HFI_PMTILES_PERMISSIONS,  # We need these to be accessible to everyone
+                Body=open(working_hfi_path, "rb"),
+            )
+            logger.info("Done uploading %s", raster_key)
+            with polygonize_in_memory(working_hfi_path, "hfi", "hfi") as layer:
+                # We need a geojson file to pass to tippecanoe
+                temp_geojson = write_geojson(layer, temp_dir)
+
+                pmtiles_filename = f'hfi{for_date.strftime("%Y%m%d")}.pmtiles'
```

Comment:
Can we use the `get_pmtiles_filename()` function from `hfi_filepath.py`?

Reply:
Oops thanks, done in https://github.com/bcgov/wps/pull/3807/commits/878183efddef8c1a4757bfa4d324207a9acfacd3